client/finality-grandpa/src/lib.rs

```diff
@@ -448,13 +450,25 @@ pub struct LinkHalf<Block: BlockT, C, SC> {
 	select_chain: SC,
 	persistent_data: PersistentData<Block>,
 	voter_commands_rx: TracingUnboundedReceiver<VoterCommand<Block::Hash, NumberFor<Block>>>,
+	justification_sender: GrandpaJustificationSubscribers<Block>,
+	justification_receiver: GrandpaJustifications<Block>,
 }
 
 impl<Block: BlockT, C, SC> LinkHalf<Block, C, SC> {
 	/// Get the shared authority set.
 	pub fn shared_authority_set(&self) -> &SharedAuthoritySet<Block::Hash, NumberFor<Block>> {
 		&self.persistent_data.authority_set
 	}
+
+	/// Get the sender end of justification notifications.
+	pub fn justification_sender(&self) -> GrandpaJustificationSubscribers<Block> {
+		self.justification_sender.clone()
+	}
+
+	/// Get the receiving end of justification notifications.
+	pub fn justification_receiver(&self) -> GrandpaJustifications<Block> {
```

Comment:
```suggestion
	pub fn justification_stream(&self) -> GrandpaJustificationStream<Block> {
```
IMHO that will be more inline with other stream methods that we have in `Client` for instance.